src/ralph/backends/data/fs.py

```diff
@@ -0,0 +1,274 @@
+
+        if not paths:
+            logger.info("No file found for query: %s", target / query.query_string)
+            return
+
+        logger.debug("Reading matching files: %s", paths)
+
+        for path in paths:
+            with path.open("rb") as file:
+                reader = self._read_raw if raw_output else self._read_dict
+                for chunk in reader(file, chunk_size):
+                    yield chunk
+
+            # The file has been read, add a new entry to the history.
+            self.append_to_history(
+                {
+                    "backend": self.name,
+                    "command": "fetch",
+                    # WARNING: previously only the file name was used as the ID
+                    # By changing this to the full file path, previously fetched files
+                    # will not be marked as read anymore.
+                    "id": str(path),
+                    "filename": path.name,
+                    "size": path.stat().st_size,
+                    "fetched_at": now(),
+                }
+            )
+
+    def write(  # pylint: disable=too-many-arguments
+        self,
+        stream: Iterable[bytes],
+        target: Union[str, None] = None,
+        chunk_size: Union[int, None] = None,
+        ignore_errors: bool = False,
+        operation_type: Union[BaseOperationType, None] = None,
+    ) -> int:
+        """Write stream records to the target file and return their count.
+
+        Args:
+            stream: (Iterable): The stream containing data to write.
+            target (str or None): The target file path.
+                If target is a relative path, it is considered to be relative to the
+                    `default_directory_path`.
+                If target is `None`, a random (uuid4) file is created in the
+                    `default_directory_path` and used as the target instead.
+            chunk_size (int or None): Ignored. Is equal to the stream chunk size.
+            ignore_errors (bool): Ignored.
+            operation_type (BaseOperationType or None): The mode of the write operation.
+                If operation_type is `CREATE` or `INDEX`, the target file is expected to
+                    be absent. If the target file exists a `FileExistsError` is raised.
+                If operation_type is `UPDATE`, the target file is overwritten.
+                If operation_type is `APPEND`, the stream content is appended to the
+                    end of the target file.
+
+        Returns:
+            int: The number of written files. (Always equal to 1).
+        """
+
+        if not operation_type:
+            operation_type = self.default_operation_type
+
+        if operation_type == BaseOperationType.DELETE:
+            msg = "Delete operation_type is not allowed."
+            logger.error(msg)
+            raise BackendParameterException(msg)
+
+        if not target:
+            target = str(uuid4())
```

Comment:
:thinking: I hesitate to write to `stdout` if `target` is missing...

If we choose to write to a random file name, shouldn't we prefix it with something like the following:

```suggestion
            target = f"{now()}-{uuid4()}"
```

Reply:
Good Idea) Thanks)